lib/view_component/base.rb

```diff
@@ -179,8 +179,9 @@ def short_identifier
       end
 
       def inherited(child)
-        # If we're in Rails, add application url_helpers to the component context
-        if defined?(Rails)
+        # If Rails application is loaded, add application url_helpers to the component context
+        # we need to check this to use this gem as a dependency
+        if Rails.application
```

Comment:
cc @dpaola2 since you opened https://github.com/github/view_component/pull/260

Reply:
indeed `defined?(Rails)` is not necessary if we create a gem as an Engine. Looking at the gem code https://github.com/dpaola2/bootstrap-viewcomponent created by @dpaola2 , this gem is not an engine. So, I believe that we can remove this condition. What do you think @joelhawksley @BlakeWilliams @rmacklin?

Reply:
> this gem is not an engine. So, I believe that we can remove this condition.

I'm not sure I understand: are you suggesting that `view_component` should only support being used in a gem which is a Rails engine? I would think that might be more restrictive than necessary.

> I can remove this condition or close this PR.

To clarify, I wasn't suggesting closing this PR; rather I was suggesting that perhaps the code should check _both_ `defined?(Rails)` and `Rails.application`. It sounded like you were running into an issue where this line raised an exception because in your case `Rails` was defined but `Rails.application` was `nil`. If we check both, I think we'd avoid an error in both @dpaola2's case and in your case.

Reply:
I'm 👍 on checking `defined?(Rails) && Rails.application`. I don't think we should require a Rails engine for a gem to provide view components.